src/main/java/org/spongepowered/mod/SpongeMod.java

```diff
@@ -166,14 +183,31 @@ public void onPreInit(FMLPreInitializationEvent e) {
 
     @Subscribe
     public void onInitialization(FMLInitializationEvent e) {
-        this.registry.init();
+        try {
+            this.registry.init();
+            if (!this.game.getServiceManager().provide(PermissionService.class).isPresent()) {
+                try {
+                    this.game.getServiceManager().setProvider(this, PermissionService.class, new SpongePermissionService());
+                } catch (ProviderExistsException e1) {
+                    // It's a fallback, ignore
+                }
+            }
+        } catch (Throwable t) {
+            logger.error("On init", t);
```

Comment:
Before the change loading continued on error, but gave a line indicating that an error occurred without printing a stacktrace or any useful info. I think the appropriate place to deal with this properly might be FML.